lib/metadata/wrapper.js

```diff
@@ -117,9 +117,9 @@ const metadata = {
         });
     },
 
-    listObject: (bucketName, prefix, marker, delimiter, maxKeys, log, cb) => {
+    listObject: (bucketName, listingParams, log, cb) => {
```

Comment:
It would be a nightmare for readability. There is [List Objects Version 2](http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html?shortFooter=true) with its own set of parameters coming as well. I will keep this unless there are more requests like yours (consensus).